### PR TITLE
Add Countries To The Address In An Invoice

### PR DESCRIPTION
**Pull request by sampoder**
I can't work out where we use this address but this should let you send invoices internationally!

### CODE REVIEW

**Comment by maxwofford** [comment]
Has this been tested? What entries does Stripe support for this?

**Comment by sampoder** [comment]
That's a good question... I tried to find where we pass this to Stripe but I don't think we even do...

**Comment by sampoder** [comment]
(that said, I may be going insane)

**Comment by cjdenio** [comment]
@sampoder it's passed to Stripe here:
https://github.com/hackclub/hcb/blob/b56bd0f1f0933eb0c98b0ae99f61da4cb139a68a/app/models/sponsor.rb#L130-L136

**Comment by sampoder** [comment]
Oh perfect! Thank you Caleb!!

File: db/migrate/20230828203220_add_country_to_sponsor.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddCountryToSponsor < ActiveRecord::Migration[7.0]
+  def change
+    add_column :sponsors, :address_country, :text, default: "United States of America"
```
Comment (cjdenio) [inline db/migrate/20230828203220_add_country_to_sponsor.rb]:
```suggestion
    add_column :sponsors, :address_country, :text, default: "US"
```

File: app/views/invoices/_form.html.erb [hunk 2]
```diff
@@ -52,9 +52,15 @@
...
-      <%= form.label :address_postal_code, "ZIP" %>
-      <%= form.text_field :address_postal_code, placeholder: "94129", required: true, disabled: @event.demo_mode? %>
+    <div class="flex">
+      <div class="field flex-auto mr1">
+        <%= form.label :address_postal_code, "ZIP" %>
+        <%= form.text_field :address_postal_code, placeholder: "94129", required: true, disabled: @event.demo_mode? %>
+      </div>
+      <div class="field flex-auto ml1">
+        <%= form.label :address_country, "Country" %>
+        <%= form.collection_select :address_country, Event.countries.map{ |c| [c.first, get_country_name_from_alpha2(c.first)] }, :first, :last, { include_blank: "Select a country", required: true, disabled: @event.demo_mode? } %>
```
Comment (cjdenio) [inline app/views/invoices/_form.html.erb]:
```suggestion
        <%= form.country_select :address_country, ["US"], priority_countries_divider: "", include_blank: "Select a country", required: true, disabled: @event.demo_mode? %>
```

File: app/views/sponsors/_form.html.erb [hunk 3]
```diff
@@ -43,6 +43,11 @@
     <%= form.text_field :address_postal_code, placeholder: "94129" %>
   </div>
 
+  <div class="field">
+    <%= form.label :address_country, "Country" %>
+    <%= form.collection_select :address_country, Event.countries.map{ |c| [c.first, get_country_name_from_alpha2(c.first)] }, :first, :last, { include_blank: "Select a country", required: true, disabled: @event.demo_mode? } %>
+  </div>
+
```
Comment (cjdenio) [inline app/views/sponsors/_form.html.erb]:
```suggestion
```

This form is deprecated and isn't shown in the UI

File: db/migrate/20230828203220_add_country_to_sponsor.rb (hunk 1 above)
Comment (cjdenio) [inline db/migrate/20230828203220_add_country_to_sponsor.rb]:
```suggestion
    add_column :sponsors, :address_country, :text, default: "US"
```

File: spec/services/invoice_service/create_spec.rb [hunk 4]
```diff
@@ -19,6 +19,7 @@
   let(:sponsor_address_city) { "Santa Monica" }
   let(:sponsor_address_state) { "CA" }
   let(:sponsor_address_postal_code) { "90401" }
+  let(:sponsor_address_country) { "United States of America" }
```
Comment (cjdenio) [inline spec/services/invoice_service/create_spec.rb]:
```suggestion
  let(:sponsor_address_country) { "US" }
```

**Review by cjdenio — CHANGES_REQUESTED** [review]

File: app/views/sponsors/_form.html.erb (hunk 3 above)
Comment (sampoder) [reply app/views/sponsors/_form.html.erb]: coolio!

**Comment by sampoder** [comment]
@cjdenio: I think Stripe wants us to pass them the two digit codes for countries. At least based on https://stripe.com/docs/api/customers/object#customer_object-shipping-address-country

**Comment by cjdenio** [comment]
That's correct, I think this code already does that?

**Comment by sampoder** [comment]
oh i really need to sleep, good catch!

**Comment by cjdenio** [comment]
lol all good!

**Comment by sampoder** [comment]
alrighty, merged in everything!

**Review by cjdenio — APPROVED** [review]
thanks!

**Pull request merged**